lib/linter/linter.js

```diff
@@ -432,10 +433,19 @@ function getDirectiveComments(filename, ast, ruleMapper, warnInlineConfig) {
 
 /**
  * Normalize ECMAScript version from the initial config
- * @param  {number} ecmaVersion ECMAScript version from the initial config
+ * @param {number} ecmaVersion ECMAScript version from the initial config
+ * @param {string} parserName the parser's name
  * @returns {number} normalized ECMAScript version
  */
-function normalizeEcmaVersion(ecmaVersion) {
+function normalizeEcmaVersion(ecmaVersion, parserName) {
+    if (parserName === DEFAULT_PARSER_NAME || parserName === void 0) {
```

Comment:
Setting the parser name should definitely not affect the ecmaVersion even after upgrading Espree. If that’s the case in this PR, we should fix that.

Reply:
I just took a deeper look at this. There’s no reason to check the parser name in this function. We just remove the check and I think things will work correctly.

Reply:
it will be a breaking change to the normalize a 3rd parser (e.g. @typescript-eslint/parser) ecmaVersion: `undefined` => 5. 

as @mdjermanovic mentioned here: https://github.com/eslint/eslint/pull/14622#discussion_r640580610

Reply:
it's set in `parserOptions` though.

Reply:
@aladdin-add we want this to be non-breaking.

@ljharb thats true and that value is still used in a bunch of places outside of the parser for legacy reasons. The new config system fixes this but for eslintrc we need to live with this weirdness.